### DISCUSSION TITLE
Pipelines instead of standalone python code, can also be a simple API Base URL

### DISCUSSION THREAD

**Discussion by Anindyadeep** [open-webui/open-webui#7838] [Ideas]
# Feature Request

As of now from what I see to add a custom pipeline, we need to either upload a python file or we need to paste the python file link of a github url. However this limits the freedom of developer. 
One solution that came to my mind could be just change the base url. 

For example:

By default OpenWeb UI might use some internal URL to let itself chat with the URL, but what if let the user to change the chat url and re-direct it to their URL (with same endpoints as the internal URL) then it could give user much more freedom of customizibility. 

Because now internally the backend would call the custom API endpoints instead of running a file. For example, suppose someone has a some very big research around LLMs, but they want to interface it with OpenWebUI's interface. 

Writing a Pipeline might become super big and messy. So What do you think about this solution? I mean this can be added with the upload / paste pipeline file?

Let me know if I am missing anything. Thanks 

## Important Notes

- **Before submitting a report**: Please check the Issues or Discussions section to see if a similar issue or feature request has already been posted. It's likely we're already tracking it! If you’re unsure, start a discussion post first. This will help us efficiently focus on improving the project.

- **Collaborate respectfully**: We value a constructive attitude, so please be mindful of your communication. If negativity is part of your approach, our capacity to engage may be limited. We’re here to help if you’re open to learning and communicating positively. Remember, Open WebUI is a volunteer-driven project managed by a single maintainer and supported by contributors who also have full-time jobs. We appreciate your time and ask that you respect ours.

- **Contributing**: If you encounter an issue, we highly encourage you to submit a pull request or fork the project. We actively work to prevent contributor burnout to maintain the quality and continuity of Open WebUI.

- **Bug reproducibility**: If a bug cannot be reproduced with a `:main` or `:dev` Docker setup, or a pip install with Python 3.11, it may require additional help from the community. In such cases, we will move it to the "issues" Discussions section due to our limited resources. We encourage the community to assist with these issues. Remember, it’s not that the issue doesn’t exist; we need your help!

Note: Please remove the notes above when submitting your post. Thank you for your understanding and support!

---

**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by gabriel-ecegi** [comment]
Hi, 
can Pipelines server integration solve your problems? 

https://docs.openwebui.com/pipelines/

It can be deployed and maintained separately.
